library/src/androidTest/java/IntegrationTests.kt

```diff
@@ -234,4 +239,35 @@ class IntegrationTests {
         ops.stopPollingOperations()
         Assert.assertFalse(ops.isPollingOperations())
     }
+
+    @Test // the z to make sure the test runs last
+    fun zTestOperationHistory() {
+        // lets create 1 operation and leave it in the state of "pending"
+        IntegrationUtils.createOperation(IntegrationUtils.Companion.Factors.F_2FA)
+        val auth = PowerAuthAuthentication()
+        auth.usePossession = true
+        auth.usePassword = pin
+        val future = CompletableFuture<List<OperationHistoryEntry>?>()
+        ops.getHistory(auth, object : IGetHistoryListener {
+            override fun onSuccess(operations: List<OperationHistoryEntry>) {
+                future.complete(operations)
+            }
+
+            override fun onError(error: ApiError) {
+                future.complete(null)
+            }
+        })
+
+        val operations = future.get(20, TimeUnit.SECONDS)
+        Assert.assertNotNull("Operations not retrieved" ,operations)
+        if (operations == null) {
+            return
+        }
+
+        Assert.assertEquals(operations.count(), 1 + operationsApproved + operationsRejected)
+        Assert.assertEquals(operations.count { it.status == OperationHistoryEntryStatus.APPROVED }, operationsApproved)
```

Comment:
Correct me if I'm wrong, but you seem to test against global variable that is set by a different unit test. This is not a good test design. Unit tests should be independent.

Reply:
You're right, the original idea was to build/test the defined lifecycle of the powerauth activation process, but it's not such a good idea.

I added a commit that makes every integration test autonomous.